src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/OrganisationCreationRequestValidator.java

```diff
@@ -60,7 +60,7 @@ public static boolean contains(String status) {
 
     public void validateOrganisationIdentifier(String inputOrganisationIdentifier) {
         if (null == inputOrganisationIdentifier || LENGTH_OF_ORGANISATION_IDENTIFIER != inputOrganisationIdentifier.length() || !inputOrganisationIdentifier.matches(ORGANISATION_IDENTIFIER_FORMAT_REGEX)) {
-            throw new EmptyResultDataAccessException(1);
+            throw new InvalidRequest("The given organisationIdentifier must be 7 Alphanumeric Characters");
         }
```

Comment:
should not this throw a specific error message based on -
1. if organisation is null or more than 7 characters long
2. organisation name is not valid

Reply:
Yes we can do that, I will add this.

Reply:
Addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/442/commits/a46046fb012a380bc963147eba1c7f3fea65fd22